telegram/ext/_commandhandler.py

```diff
@@ -126,6 +137,39 @@ def __init__(
             filters if filters is not None else filters_module.UpdateType.MESSAGES
         )
 
+        self.has_args = has_args
+
+    def _check_correct_args(self, args: List[str]) -> Optional[bool]:
+        class ArgsCondition(Enum):
+            HAS_ARGS_TRUE_ARGS_PRESENT = auto()
+            HAS_ARGS_FALSE_ARGS_ABSENT = auto()
+            HAS_ARGS_NUM = auto()
+            HAS_ARGS_NONE = auto()
+            HAS_INVALID_ARGS = auto()
+
+        mapping = {
+            ArgsCondition.HAS_ARGS_TRUE_ARGS_PRESENT: True,
+            ArgsCondition.HAS_ARGS_FALSE_ARGS_ABSENT: True,
+            ArgsCondition.HAS_ARGS_NONE: True,
+            ArgsCondition.HAS_ARGS_NUM: True,
+            ArgsCondition.HAS_INVALID_ARGS: False,
+        }
+
+        condition = (
+            ArgsCondition.HAS_ARGS_NONE
+            if (self.has_args is None)
+            else ArgsCondition.HAS_ARGS_TRUE_ARGS_PRESENT
+            if (self.has_args is True and args)
+            # Must specify is True, otherwise if has_args is int, it will default to Truthy value.
+            else ArgsCondition.HAS_ARGS_FALSE_ARGS_ABSENT
+            if (self.has_args is False and not args)
+            else ArgsCondition.HAS_ARGS_NUM
+            if (isinstance(self.has_args, int) and len(args) == self.has_args)
+            else ArgsCondition.HAS_INVALID_ARGS
+        )
```

Comment:
The enum is a nice idea, but it seems like rather heavy overkill for this use case :D especially since there are only 2 values that the enums are mapped to.
The important part are the if-clauses you specified and those look good to me. I suggest to simplify this method to something like this:

```python
if (cond 1) or (cond 2) or (cond 3) or (cond 4):
    return True
return False
```